runtime/utils/common/src/aws/smithy/kotlin/runtime/util/Platform.kt

```diff
@@ -9,16 +9,41 @@ package aws.smithy.kotlin.runtime.util
 public expect object Platform {
     /**
      * Get an environment variable or null
+     *
+     * @param key name of environment variable
+     * @return value of environment variable or null if undefined
      */
     fun getenv(key: String): String?
 
+    /**
+     * Get a system property (on supported platforms)
+     *
+     * @param key name of environment variable
+     * @return value of system property or null if undefined or platform does not support properties
+     */
+    fun getProperty(key: String): String?
+
     val isJvm: Boolean
     val isAndroid: Boolean
     val isBrowser: Boolean
     val isNode: Boolean
     val isNative: Boolean
 
+    /**
+     * The delimiter of segments in a path. For example in Linux: /home/user/documents
+     * or Windows: C:\Program Files\Notepad.EXE
+     */
+    val filePathSegment: String
```

Comment:
nit

`filePathSeparator` seems more appropriate to me

Reply:
sure, updated